### PR TITLE
Ignore paths in the scanner with `.ignore` or `.plexignore` files

### PR DESCRIPTION
**Pull request by niamu**
This change supports `.ignore` and `.plexignore` files within scanned media directories as described in [Plex's documentation](https://support.plex.tv/articles/201381883-special-keyword-file-folder-exclusion/#toc-0).

This is achieved via the [ignore](https://docs.rs/ignore/latest/ignore/struct.WalkBuilder.html) library which means it will also be affected by `.gitignore` files by default. Possibly something we should intentionally exclude?

[Related issue](https://github.com/Dusk-Labs/dim/issues/520)

### CODE REVIEW

File: dim/src/scanner/mod.rs [hunk 1]
```diff
@@ -41,11 +42,65 @@ use tracing::error;
 use tracing::info;
 use tracing::instrument;
 use tracing::warn;
-use walkdir::WalkDir;
 
 pub use error::Error;
 
-pub(super) static SUPPORTED_EXTS: &[&str] = &["mp4", "mkv", "avi", "webm"];
+pub(super) static SUPPORTED_EXTS: &[&str] = &[
```
Comment (niamu) [inline dim/src/scanner/mod.rs:48]: In testing this feature I realized there were several media files that Dim wasn't picking up so I copied this list from Jellyfin which seems comprehensive, but may also be considered overkill. https://github.com/jellyfin/jellyfin/blob/master/Emby.Naming/Common/NamingOptions.cs#L24-L80

Comment (mental32) [reply dim/src/scanner/mod.rs:48]: little overkill, 100% for it

File: dim-core/src/scanner/mod.rs [hunk 2]
```diff
@@ -41,11 +42,65 @@ use tracing::error;
 use tracing::info;
 use tracing::instrument;
 use tracing::warn;
-use walkdir::WalkDir;
 
 pub use error::Error;
 
-pub(super) static SUPPORTED_EXTS: &[&str] = &["mp4", "mkv", "avi", "webm"];
+pub(super) static SUPPORTED_EXTS: &[&str] = &[
```
Comment (vgarleanu) [inline dim-core/src/scanner/mod.rs:48]: Oh boy, not a fan of how big this is now. Can we maybe move it to a file or a separate module and not inline it?

**Review by vgarleanu — COMMENTED** [review]
LGTM

File: dim-core/src/scanner/mod.rs (hunk 2 above)
Comment (niamu) [reply dim-core/src/scanner/mod.rs:48]: I'm happy to move this elsewhere. If you have suggestions for where that file or module should be located and named let me know and I'll make the adjustment.

Comment (mental32) [reply dim-core/src/scanner/mod.rs:48]: We can always tidy it later, good-enough for now :)

**Comment by mental32** [comment]
Same thing with the other PR, you'll have to rebase

**Comment by niamu** [comment]
I have rebased. Is there something I missed or didn't move correctly?

**Pull request merged**